### PR TITLE
Add individual WhiteList entry deletion

### PR DESCRIPTION
**Pull request by tusharsnn**
## Description
This PR will add an individual WhiteList entry deletion feature to the extension's `/options` page. User will be able to delete specific whitelist domain from the full list of whitelist domains visible on the page. 

Closes: #78

### CODE REVIEW

**Comment by tusharsnn** [comment]
Consider merging #105 before this, as the PR branch is based off of the earlier PR.

**Comment by immattdavison** [comment]
> Consider merging #105 before this, as the PR branch is based off of the earlier PR.

Definitely! Will test these PRs tonight!

File: assets/cross.svg [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="iso-8859-1"?>
```
Comment (immattdavison) [inline assets/cross.svg:1]: This works but probably better to use a unicode char

File: options.js [hunk 2]
```diff
@@ -130,21 +118,118 @@ function removeWhitelist(){
...
+        // no whitelist entry to save
+        await chrome.storage.local.remove(["user_whitelist", "whiteList_RuleIds"]);
+        whiteList_domains_table.style.display = "none";
+        Erase_Button.style.display = "none";
+    }
 }
+
 document.addEventListener('DOMContentLoaded', restore_options);
 
 WhiteList_Button.addEventListener("click", addWhiteList);
```
Comment (immattdavison) [inline options.js:235]: lgtm

**Review by immattdavison — APPROVED** [review]

**Comment by immattdavison** [comment]
@tusharsnn after testing this does not actually remove the whitelist entry

**Comment by tusharsnn** [comment]
what issue did you face ? I tested it and seems to work fine.

Can you give me steps to reproduce the issue that you are facing?

**Comment by immattdavison** [comment]
> what issue did you face ? I tested it and seems to work fine.
> 
> Can you give me steps to reproduce the issue that you are facing?

Upon clicking the delete button nothing changed and the domain remained part of the whitelist.

**Comment by tusharsnn** [comment]
I think I understand the issue, the "remove all domains" doesn't work as expected with the new modifications. To solve this, we need to use async/await instead of callbacks. Should I go improve it a bit if you like ? 
The current implementation will also likely improve as async await is easier to read and maintain and reduces code lines.

**Comment by immattdavison** [comment]
> I think I understand the issue, the "remove all domains" doesn't work as expected with the new modifications. To solve this, we need to use async/await instead of callbacks. Should I go improve it a bit if you like ? 
> 
> The current implementation will also likely improve as async await is easier to read and maintain and reduces code lines.

When I ran tests in the live version with this merged the `X` buttons didn't work. 

This is the issue.

Feel free to make changes as you think will fix it, I look forward to seeing your solution.

**Pull request merged**